spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,429 @@
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.spark.k8s.operator.reconciler.SparkReconcilerUtils;
+
+/** Spark Operator Configuration options. */
+@Slf4j
+public class SparkOperatorConf {
+  public static final ConfigOption<String> OPERATOR_APP_NAME =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.name")
+          .typeParameterClass(String.class)
+          .description("Name of the operator.")
+          .defaultValue("spark-kubernetes-operator")
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_NAMESPACE =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.namespace")
+          .typeParameterClass(String.class)
+          .description("Namespace that operator is deployed within.")
+          .defaultValue("default")
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<String> OPERATOR_WATCHED_NAMESPACES =
+      ConfigOption.<String>builder()
+          .key("spark.kubernetes.operator.watchedNamespaces")
+          .description(
+              "Comma-separated list of namespaces that the operator would be "
+                  + "watching for Spark resources. If unset, operator would "
+                  + "watch all namespaces by default.")
+          .defaultValue(null)
+          .typeParameterClass(String.class)
+          .build();
+
+  public static final ConfigOption<Boolean> TERMINATE_ON_INFORMER_FAILURE =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.terminateOnInformerFailure")
+          .typeParameterClass(Boolean.class)
+          .description(
+              "Enable to indicate informer errors should stop operator startup. If "
+                  + "disabled, operator startup will ignore recoverable errors, "
+                  + "caused for example by RBAC issues and will retry "
+                  + "periodically.")
+          .defaultValue(false)
+          .enableDynamicOverride(false)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_TERMINATION_TIMEOUT_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.terminationTimeoutSeconds")
+          .description(
+              "Grace period for operator shutdown before reconciliation threads are killed.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(30)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.parallelism")
+          .description(
+              "Thread pool size for Spark Operator reconcilers. Use -1 for unbounded pool.")
+          .enableDynamicOverride(false)
+          .typeParameterClass(Integer.class)
+          .defaultValue(30)
```

Comment:
May I ask why do we have this default value, `30`?

Reply:
it's an experimental number yield from integration test while migrating to josdk. For default count, learning from flink operator we updated that to `50`